### PR TITLE
[JUJU-2287] Add secret provider config schema and validation

### PR DESCRIPTION
**Pull request by wallyworld**
Secret providers now support a schema for their config attributes.
The schema is used to validate config when creating a backend and also to understand which config attributes are secret and should not be displayed by default when listing backends.

So far, this is implemented for vault. k8s backends just use the current model cluster, but we will need config when we support external k8s clusters.

Also remove unseal keys used for developers.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

bootstrap on lxd and add a vault backend

```
cat /path/to/vaultcfg.yaml
endpoint: "http://10.0.0.77:8200"
token: "hvs.OuTFPv7qV1AiyXYfwkDYRMwv"
unseal-keys:
  - r/QmaheRvVVqppU78t3cCGZnIealuGPmWc6JBfo7Gjs=

juju add-secret-backend myvault vault --config /path/to/vaultcfg.yaml
juju model-config secret-backend=myvault
juju exec --unit controller/0 "secret-add foo=bar"
secret:ce4kvahk9k63k9kjm4rg

juju show-secret secret:ce4kvahk9k63k9kjm4rg --reveal
ce4kvahk9k63k9kjm4rg:
  revision: 1
  owner: controller
  created: 2022-12-02T01:06:51Z
  updated: 2022-12-02T01:06:51Z
  content:
    foo: bar

juju secret-backends --format yaml --reveal
internal:
  name: internal
  backend: controller
  secrets: 666
myvault:
  name: myvault
  backend: vault
  config:
    endpoint: http://10.0.0.77:8200
    token: hvs.OuTFPv7qV1AiyXYfwkDYRMwv
    unseal-keys:
    - r/QmaheRvVVqppU78t3cCGZnIealuGPmWc6JBfo7Gjs=
  secrets: 666

```
check the secret is stored in vault

switch the backend
```
juju model-config secret-backend=internal
```
add a secret and get the value; check the secret is stored in mongo

do the same on a k8s model, noting that no backend config is needed - it just uses k8s out of the box.

### CODE REVIEW

File: secrets/provider/vault/config.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	ClientCertKey: {
+		Description: "The vault client certificate.",
+		Type:        environschema.Tstring,
+	},
+	ClientKeyKey: {
+		Description: "The vault client certificate key.",
+		Type:        environschema.Tstring,
+		Secret:      true,
+	},
+	UnsealKeysKey: {
```
Comment (tlm) [inline secrets/provider/vault/config.go:57]: It's my belief that we should not be supporting the storage of Vault UnsealKeys in Juju at all. This is the fundamental encryption layer of vault and Juju is not in any position to store this information. Even the hint that Juju supports this has the potential to encourage users to take on practices that are not safe or implies that Juju can be a trusted source for this information.

For more context https://developer.hashicorp.com/vault/docs/concepts/seal

If this is to aid in development where a normal `vault -dev` mode will not suffice then we should build adequate tooling and scripts into our build pipeline to help us but including this in a production release stinks on Juju's part.

We try to promote our view on best practice for modeling the world and how to manage operations. I don't think this fits that narrative except to make a developers life easier for 2 seconds.

Comment (wallyworld) [reply secrets/provider/vault/config.go]: they were just there for developers, removed now

**Review by hpidcock — APPROVED** [review]
LGTM thanks

**Pull request merged**